pkg/storage/gcp/gcp.go

```diff
@@ -3,40 +3,48 @@ package gcp
 import (
 	"context"
 	"fmt"
+	"net/http"
 	"net/url"
 
 	"cloud.google.com/go/storage"
 	"github.com/gomods/athens/pkg/config/env"
-	"google.golang.org/api/option"
+	"github.com/gomods/athens/pkg/errors"
+	"google.golang.org/api/googleapi"
 )
 
 // Storage implements the (./pkg/storage).Backend interface
 type Storage struct {
 	bucket       Bucket
 	baseURI      *url.URL
 	closeStorage func() error
+	projectID    string
 }
 
-// NewWithCredentials returns a new Storage instance authenticated using the provided
+// New returns a new Storage instance authenticated using the provided
 // ClientOptions. The bucket name to be used will be loaded from the
 // environment variable ATHENS_STORAGE_GCP_BUCKET.
 //
-// The ClientOptions should provide permissions sufficient to read, write and
-// delete objects in google cloud storage for your project.
-func NewWithCredentials(ctx context.Context, cred option.ClientOption) (*Storage, error) {
-	storage, err := storage.NewClient(ctx, cred)
+// If you're on your dev machine, make sure you're logged in to gcloud.
+// If you're outside GCP, make sure you have GOOGLE_APPLICATION_CREDENTIALS set.
```

Comment:
On a dev machine, setting `GOOGLE_APPLICATION_CREDENTIALS` should be enough. Can link to https://cloud.google.com/docs/authentication/getting-started for more detail.

Suggested:
```
// If you're not running on GCP, set the GOOGLE_APPLICATION_CREDENTIALS environment variable
// to the path of your service account file. If you're running on GCP (e.g. AppEngine),
// credentials will be automatically provided.
// See https://cloud.google.com/docs/authentication/getting-started.
```